sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -43,47 +44,77 @@
+    if check_case:
+        shared_access_key = conn_settings.get("SharedAccessKey")
+        shared_access_key_name = conn_settings.get("SharedAccessKeyName")
+        endpoint = conn_settings.get("Endpoint")
+        entity_path = conn_settings.get("EntityPath")
+        # non case sensitive check when parsing connection string for internal use
+        for key, value in conn_settings.items():
+            # only sas check is non case sensitive for both conn str properties and internal use
+            if key.lower() == "sharedaccesssignature":
+                shared_access_signature = value
+
+    if not check_case:
+        endpoint = conn_settings.get("endpoint") or conn_settings.get("hostname")
+        if endpoint:
+            endpoint = endpoint.rstrip("/")
+        shared_access_key_name = conn_settings.get("sharedaccesskeyname")
+        shared_access_key = conn_settings.get("sharedaccesskey")
+        entity_path = conn_settings.get("entitypath")
+        shared_access_signature = conn_settings.get("sharedaccesssignature")
+
+    if shared_access_signature:
+        try:
+            # Expiry can be stored in the "se=<timestamp>" clause of the token. ('&'-separated key-value pairs)
+            shared_access_signature_expiry = int(
+                shared_access_signature.split("se=")[1].split("&")[0]   # type: ignore
+            )
+        except (
+            IndexError,
+            TypeError,
+            ValueError,
+        ):  # Fallback since technically expiry is optional.
+            # An arbitrary, absurdly large number, since you can't renew.
+            shared_access_signature_expiry = int(time.time() * 2)
+
+    entity = cast(str, eventhub_name or entity_path)
+
+    # check that endpoint is valid
+    if not endpoint:
+        raise ValueError("Connection string is either blank or malformed.")
+    parsed = urlparse(endpoint)
+    if not parsed.netloc:
+        raise ValueError("Invalid Endpoint on the Connection String.")
+    host = cast(str, parsed.netloc.strip())
+
+    if any([shared_access_key, shared_access_key_name]) and not all(
+        [shared_access_key, shared_access_key_name]
+    ):
         raise ValueError(
             "Invalid connection string. Should be in the format: "
             "Endpoint=sb://<FQDN>/;SharedAccessKeyName=<KeyName>;SharedAccessKey=<KeyValue>"
         )
-    entity = cast(str, eventhub_name or entity_path)
-    left_slash_pos = cast(str, endpoint).find("//")
-    if left_slash_pos != -1:
-        host = cast(str, endpoint)[left_slash_pos + 2 :]
-    else:
-        host = str(endpoint)
+    if shared_access_signature and shared_access_key:
+        raise ValueError(
+            "Only one of the SharedAccessKey or SharedAccessSignature must be present."
+        )
+    if not shared_access_signature and not shared_access_key:
+        raise ValueError(
+            "At least one of the SharedAccessKey or SharedAccessSignature must be present."
+        )
```

Comment:
If I'm understanding correctly, `def _parse_conn_str(conn_str, **kwargs):` is now the shared method used by 
1. connection string parser
2. client construction
while previously we had two separate part.

but this part of logic seems to only lies in the connection string parser before, so I think is this actually a breaking change here to the user?

example:
let's say user has a connection string:
`Endpoint=....;SharedAccessKeyName=...;SharedAccessKey=...;SharedAccessSignature=;`
it would work in `EventHubProducerClient.from_connection_string(...)` previously while after the change user would get ValueError.

Reply:
yes, I think that's true for line 109, but line 113 previously raised ValueError too (in the case that neither sas nor key was included). Since only the connection string parser logic requires the check for line 109 and it checks case, does this make sense to you for line 109?
`if check_case and shared_access_signature and shared_access_key:`

Reply:
or I can move the check from 109 to 77.

Reply:
I'm good with `if check_case and shared_access_signature and shared_access_key` as long as we don't break users. ( I don't think moving from 109 to 77 should matter?)

probably adding a comment saying that connection string parser (in which case check_case is True) needs checking shared_access_signature and shared_access_key while client construction should not have this for backward compatibility.